lib/hydra/works/models/concerns/generic_file/characterization.rb

```diff
@@ -0,0 +1,120 @@
+  module Characterization
+    extend ActiveSupport::Concern
+
+    autoload :FitsDatastream, 'hydra/works/models/generic_file/characterization/fits_datastream.rb'
+    autoload :Image, 'hydra/works/models/concerns/generic_file/characterization/image.rb'
+    autoload :Doc, 'hydra/works/models/concerns/generic_file/characterization/doc.rb'
+    autoload :Video, 'hydra/works/models/concerns/generic_file/characterization/video.rb'
+    autoload :Audio, 'hydra/works/models/concerns/generic_file/characterization/audio.rb'
+
+    included do
+      include Hydra::Derivatives::ExtractMetadata
+
+      # Base characterization properties
+
+      property :filename, predicate: RDF::Vocab::EBUCore.filename, multiple: false
+      property :format_label, predicate: RDF::Vocab::PREMIS.hasFormatName
+      property :file_size, predicate: RDF::Vocab::EBUCore.fileSize
+      property :well_formed, predicate: RDF::URI.new("http://projecthydra.org/ns/fits/wellFormed")
+      property :valid, predicate: RDF::URI.new("http://projecthydra.org/ns/fits/valid")
+      property :date_created, predicate: RDF::Vocab::EBUCore.dateCreated
+      property :last_modified, predicate: RDF::Vocab::EBUCore.dateModified
+      property :fits_version, predicate: RDF::Vocab::PREMIS.hasCreatingApplicationVersion
+      property :exif_version, predicate: EXIF.exifVersion
+      property :original_checksum, predicate: RDF::Vocab::PREMIS.hasMessageDigest
+      # property :mime_type, delegate_to: 'original_file'
+      property :mime_type, predicate: RDF::Vocab::EBUCore.hasMimeType, multiple: false do |index|
+        index.as :stored_searchable
+      end
+
+      # Parser config is :term => :property, and is consulted before defaulting to a
+      # property that matches the term.  Required when term and property aren't the same.
+      class_attribute :parser_config
+      self.parser_config = { exif_tool_version: :exif_version,
+                             file_mime_type: :mime_type }
+
+      class_attribute :parser_class
+      self.parser_class = FitsDatastream
+
+      # Extract the metadata from a content file and store it as properties.
+      # The target properties must already exist.
+      # @param [ActiveFedora::File] file to characterize
+      def characterize(file = original_file)
+        # Run fits and get xml string
+        extracted_md = extract_metadata(file)
+        # Parse xml string to get fits datastream terms and values
+        terms = parse_metadata(extracted_md)
+        # Store terms for which there are properties mapped.
+        store_metadata(terms)
+      end
+
+      protected
+
+      def extract_metadata(file = original_file)
+        return unless file.has_content?
+        Hydra::FileCharacterization.characterize(file.content, file.original_name, :fits) do |cfg|
+          cfg[:fits] = Hydra::Derivatives.fits_path
+        end
+      end
+
+      # Use OM to parse metadata
+      def parse_metadata(metadata)
+        fdata = parser_class.new
```

Comment:
It was originally the parsed file data, but the approach of this branch actually does use an OmDatastream to do the parsing.  It just throws it away when it's done instead of keeping it around and delegating properties to it.